src/test/app/Offer_test.cpp

```diff
@@ -795,22 +850,831 @@ class Offer_test : public beast::unit_test::suite
             owners ("eve", 1));
     }
 
+    void
+    testcaseSelfCross()
+    {
+        testcase ("Self-crossing");
+
+        using namespace jtx;
+
+        auto const gw = Account {"gateway"};
+        auto const USD = gw["USD"];
+        auto const BTC = gw["BTC"];
+
+        Env env {*this};
+        env.fund (XRP (10000), gw);
+        env (offer (gw, BTC (500), USD (100)));
+        env (offer (gw, USD (100), BTC (500)));
+        env.close ();
```

Comment:
good suggestion - I've added some explicit checks of the two books after each offer is made. Hopefully that's what you had in mind.